flocker/node/agents/blockdevice.py

```diff
@@ -871,10 +1003,11 @@ def calculate_necessary_state_changes(self, local_state,
             for manifestation
             in manifestations_to_create
         )
+        # XXX: Maybe list volumes here and pass the list to each helper below.
+        deletes = self._calculate_deletes(configured_manifestations) # volumes=volumes)
```

Comment:
Note that `_calculate_deletes` changed to not list volumes.  This is because we want to keep `calculate_necessary_state_changes` as simple as we can.  One way to do that is to only operate on the simple model objects given as inputs.  For now we're accepting some extra complexity that results from this decision in the IStateChange implementations (eg, make the resize operation a no-op if the desired size is already the actual size).  Later, we'll fix this up by making `local_state` and `cluster_state` carry all of the extra information we need to make smarter decisions.